src/sys/socket/consts.rs

```diff
@@ -1,75 +1,243 @@
 pub use self::os::*;
 
-#[cfg(any(target_os = "linux", target_os = "android"))]
 mod os {
     use libc::{self, c_int, uint8_t};
 
     pub const AF_UNIX: c_int  = libc::AF_UNIX;
     pub const AF_LOCAL: c_int = libc::AF_LOCAL;
     pub const AF_INET: c_int  = libc::AF_INET;
     pub const AF_INET6: c_int = libc::AF_INET6;
+    #[cfg(any(target_os = "linux", target_os = "android"))]
     pub const AF_NETLINK: c_int = libc::AF_NETLINK;
+    #[cfg(any(target_os = "linux", target_os = "android"))]
     pub const AF_PACKET: c_int = libc::AF_PACKET;
+    #[cfg(any(target_os = "macos", target_os = "ios"))]
+    pub const AF_SYSTEM: c_int = libc::AF_SYSTEM;
+    #[cfg(any(target_os = "macos", target_os = "ios"))]
+    pub const AF_SYS_CONTROL: c_int = 2;
 
     pub const SOCK_STREAM: c_int = libc::SOCK_STREAM;
     pub const SOCK_DGRAM: c_int = libc::SOCK_DGRAM;
     pub const SOCK_SEQPACKET: c_int = libc::SOCK_SEQPACKET;
     pub const SOCK_RAW: c_int = libc::SOCK_RAW;
+    #[cfg(not(any(target_os = "linux", target_os = "android")))]
+    pub const SOCK_RDM: c_int = libc::SOCK_RDM;
```

Comment:
Ahh, yes, missed that, sorry. But it is a good idea to use a whitelist and not a blacklist, as we'll be adding new platforms moving forward and they'll be more robust to these additions (though unfortunately likely more verbose).